### PR TITLE
Add API Status Page

### LINKED ISSUE #353: API status page
A page on the app (maybe policyengine.org/status) that lists each API endpoint and whether it responded to a simple input, along with the latency.

### PR DESCRIPTION
**Pull request by samau3**
Fixes #353.

Implement an API status page that makes requests to some of the API endpoints for PolicyEngine via `policyengine.org/status`. 

Note the below gif ends before the US `/calculate` request finishes, and I wasn't able to access Nigeria's Earnings Variation pages to extract a household input object. However, this allows a showcasing of all three different request statuses that the page can display. 
Demo of API Status Page

### CODE REVIEW

File: src/PolicyEngine.jsx [hunk 1]
```diff
@@ -23,6 +24,7 @@ export function PolicyEngineRoutes() {
       <Route path="/us/*" element={<PolicyEngineCountry countryId="us" />} />
       <Route path="/ca/*" element={<PolicyEngineCountry countryId="ca" />} />
       <Route path="/ng/*" element={<PolicyEngineCountry countryId="ng" />} />
+      <Route path="/status" element={<StatusPage />} />
```
Comment (nikhilwoodruff) [inline src/PolicyEngine.jsx:27]:
```suggestion
      <Route path="/api-status" element={<StatusPage />} />
```

File: src/pages/StatusPage.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        >
+          <div
+              style={{
+                paddingRight: 0,
+                paddingLeft: 0,
+              }}
+            >
+            <h1>{`PolicyEngine's API Status Information`}</h1>
+            <section>
+              <h3>Country Pages</h3>
```
Comment (nikhilwoodruff) [inline src/pages/StatusPage.jsx:81]: Generally we try to avoid Capitalising Each Word btw.

File: src/pages/StatusPage.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                      style={{ 
+                        color: STATUS_TEXT_COLORS[countryStatuses[countryStatus]["status"]],
+                        backgroundColor: STATUS_COLORS[countryStatuses[countryStatus]["status"]],
+                        maxWidth: "50%",
+                        marginTop: 2,
+                        marginLeft: "auto",
+                        marginRight: "auto"
+                      }} 
+                    >
+                        {countryStatus.toUpperCase()} - {countryStatuses[countryStatus]["status"] ? countryStatuses[countryStatus]["status"] : "Checking Status"} - {countryStatuses[countryStatus]["latency"] ? countryStatuses[countryStatus]["latency"] : "Calculating Latency"}
```
Comment (nikhilwoodruff) [inline src/pages/StatusPage.jsx:95]: Could we instead use:

United Kingdom | `/uk` | `OK` in 476ms

File: src/pages/StatusPage.jsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+              style={{
+                paddingRight: 0,
+                paddingLeft: 0,
+              }}
+            >
+            <h1>{`PolicyEngine's API Status Information`}</h1>
+            <section>
+              <h3>Country Pages</h3>
+              <h6>{`These requests represent accessing the home page for PolicyEngine's supported countries.`}</h6>
+                {Object.keys(countryStatuses).map((countryStatus, idx) =>
```
Comment (nikhilwoodruff) [inline src/pages/StatusPage.jsx:83]: Could we put this into two columns side-by-side:

Left: (text left-aligned, but still have everything in the narrow centre-block, like the blog posts)

## Country pages

These requests represent...

Right:

First result
Second result
...

File: src/pages/StatusPage.jsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                )}
+            </section>
+            <section>
+              <h3 style={{marginTop: 5}}>Metadata</h3>
+              <h6>{`These requests represent accessing the metadata for PolicyEngine's supported countries.`}</h6>
+              {Object.keys(metaDataStatuses).map((metaDataStatus, idx) => 
+                  <div 
+                    key={`${idx + 1}-metaData`} 
+                    style={{ 
+                        color: STATUS_TEXT_COLORS[metaDataStatuses[metaDataStatus]["status"]],
```
Comment (nikhilwoodruff) [inline src/pages/StatusPage.jsx:106]: Nice-to-have: could we wrap these in a functional component given the duplication?

**Review by nikhilwoodruff — COMMENTED** [review]
This is so cool @samau3. Just a few suggestions. Could we also add an _API status_ link in the homepage's footer next to the other links?

**Comment by samau3** [comment]
Here's the new status page with the changes suggested!

Updated status page

**Review by nikhilwoodruff — APPROVED** [review]
Amazing. Looks great- merging. Thanks @samau3!

**Pull request merged**